src/cf.data.pre

```diff
@@ -1892,10 +1893,13 @@ http_access deny CONNECT !SSL_ports
 http_access allow localhost manager
 http_access deny manager
 
-# We strongly recommend the following be uncommented to protect innocent
-# web applications running on the proxy server who think the only
-# one who can access services on "localhost" is a local user
-#http_access deny to_localhost
+# Protect web applications running on the same server as Squid. They often
+# assume that only local users can access them at "localhost" ports.
+http_access deny to_localhost
```

Comment:
> Amos at https://github.com/squid-cache/squid/pull/1161#issuecomment-1269273970: Also, do you really want to return Squid to the default of performing an extra DNS lookup on every request?

I do not _want_ that, but I think it is the right tradeoff for the _default_ configuration. As you know, I strive to make such decisions based on principles rather than whims of the moment. The specific principle here is [documented](https://github.com/squid-cache/squid/pull/1161#issue-1398153736) as "The default configuration should either block all traffic or only allow traffic that is unlikely to introduce new attack vectors into the network". In this context, the need for DNS queries and their performance impact is a secondary concern.

Eventually, Squid may gain a feature that would allow to achieve the same safety level without any DNS lookups. At that time, this default configuration could be adjusted to eliminate those lookups.

> You just went to all the trouble of removing the logformat %>A performance optimization to remove these early DNS costs.

It sounds like you are misinterpreting what happened in commit a8c7a11. That commit did not remove any performance optimizations. It gave the admin control over whether DNS lookups are performed while preserving the existing "early lookup" feature for virtually all use cases.

This new default configuration does not take the control restored in a8c7a11 away from the admin -- the admin should adjust the default to suit their needs, of course, including properly resolving various performance/safety tradeoffs.